dist/valid-head.js

```diff
@@ -0,0 +1,36 @@
+// Used for WPT
+//[valid-head]
+// Use DOMParser to extract head elements so browser can't truncate the head
+// Loop through each node and check against valid elements array
+// Returns object with ARRAY and BOOL
+try {
+    var html = $WPT_BODIES[0].response_body;
+    // create in interrupt element which prevents truncation.
+    html = html.replace('</head>', "<interrupt />\n</head>");
+
+    var doc = document.implementation.createHTMLDocument("New Document");
+    var head = doc.createElement("div");
+    head.innerHTML = html;
+
+    [...head.getElementsByTagName("interrupt")].forEach(el => el.remove());
+    var nodes = head.getElementsByTagName("*");
```

Comment:
Not quite following this part. You're creating a `div` element, setting its contents to be the entire response body, removing the `interrupt` element, and querying everything else. How does that entire that you're only querying elements in the `head`?

Reply:
We create an `interrupt` element just before the head closes. The rest of the HTML document ends up in the `interrupt` element and then we delete everything inside of that. 
It's like pulling a reverse uno card on the HTML trying to correct itself.

Reply:
Here is a demo showing this working on Chrome.  https://www.screencast.com/t/q12dpBfUaqy  For full disclosure, I am not 100% sure how this works, but noticed it a while ago while working on something else.

Reply:
🤷🪄

Reply:
Querying these results now 👀 

invalid | pages | n
-- | -- | --
img | 1,470,348 | 7,022,280
div | 652,648 | 21,726,625
a | 342,369 | 21,075,376
p | 212,880 | 6,538,475
span | 208,986 | 9,677,474
br | 176,658 | 5,512,635
iframe | 171,274 | 231,499
input | 159,036 | 1,800,872
ul | 141,196 | 1,683,013
li | 141,182 | 10,075,377
form | 137,864 | 401,917
h1 | 127,850 | 429,224
button | 119,402 | 533,072
h2 | 117,586 | 756,110
h3 | 103,785 | 1,045,688
svg | 101,265 | 1,106,651
footer | 97,547 | 424,299
header | 96,379 | 377,136
path | 95,030 | 1,681,346
label | 94,958 | 570,545
nav | 93,122 | 177,827
b | 92,002 | 1,774,036
strong | 83,602 | 1,734,224
table | 78,495 | 883,732
tbody | 78,239 | 879,676
tr | 78,226 | 2,687,622
i | 77,575 | 1,264,969
td | 77,514 | 5,867,209
article | 67,777 | 555,650
g | 67,408 | 867,298